rubin_scheduler/scheduler/basis_functions/mask_basis_funcs.py

```diff
@@ -194,6 +195,77 @@ def _calc_value(self, conditions, indx=None):
         return result
 
 
+class AltAzShadowMaskBasisFunction(BaseBasisFunction):
+    """Mask any out of range altitudes and azimuths, then extend the
+    mask so if observations are taken in pairs, the second in the pair will
+    not have moved into a masked region.
+    """
```

Comment:
Is this intended to be used as the basic "mask out of range values" or only in the case of "extend the mask for out of range values to include a shadow region" ? (I think it's both, and the shadow could be 0, but the docstring could be more clear. Also it needs to include the parameters and explain what shadow_minutes means.) 

Think of it as explaining the usage and purpose of this class to people who mostly know what the scheduler does, but don't know anything about the details and shouldn't have to read the code to find out.

Reply:
Oh yeah, and I guess you want to also explain what the conditions.alt_limits and conditions.az_limits do here too and how they interact with min_alt and max_alt.

Reply:
Improving this docstring would be really helpful for people using this function, particularly to know the interaction with the conditions values.